src/appleseed.studio/mainwindow/pythonconsole/pythonconsolewidget.cpp

```diff
@@ -121,14 +165,166 @@ void PythonConsoleWidget::slot_execute_all()
     execute(m_input->toPlainText());
 }
 
+void PythonConsoleWidget::execute(const QString& script)
+{
+    PythonInterpreter::instance().execute(script.toStdString().c_str());
+}
+
 void PythonConsoleWidget::slot_clear_output()
 {
     m_output->clear();
 }
 
-void PythonConsoleWidget::execute(const QString& script)
+void PythonConsoleWidget::slot_new_file()
 {
-    PythonInterpreter::instance().execute(script.toStdString().c_str());
+    if (!can_close_file())
+        return;
+
+    close_file();
+}
+
+void PythonConsoleWidget::close_file()
+{
+    m_input->clear();
+    m_opened_file = "";
+    m_is_file_dirty = false;
+}
+
+namespace
+{
+    int show_modified_file_message_box(QWidget* parent)
+    {
+        QMessageBox msgbox(parent);
+        msgbox.setWindowTitle("Save Changes?");
+        msgbox.setIcon(QMessageBox::Question);
+        msgbox.setText("The python script has been modified.");
```

Comment:
Capitalize Python everywhere: python is a snake, Python is a programming language :)